LEGO1/lego/legoomni/src/build/legocarbuild.cpp

```diff
@@ -1276,12 +1346,172 @@ void LegoCarBuild::Enable(MxBool p_enable)
 	}
 }
 
-// STUB: LEGO1 0x10025720
-// STUB: BETA10 0x1006e9df
-undefined4 LegoCarBuild::FUN_10025720(undefined4 p_param1)
+// FUNCTION: BETA10 0x10070520
+inline MxU32 LegoCarBuild::Beta0x10070520()
 {
-	// TODO
-	return 0;
+	switch (m_carId) {
+	case Helicopter_Actor:
+		return 2;
+	case DuneBugy_Actor:
+		return 0;
+	case Jetski_Actor:
+		return 1;
+	case RaceCar_Actor:
+		return 3;
+	default:
+		assert(0);
+		return 0;
+	}
+}
+
+inline void LegoCarBuild::StopActionIn0x344()
+{
+	// There is no direct evidence for this inline function in LEGO1,
+	// but some code doesn't make much sense otherwise. For example,
+	// sometimes `m_unk0x344` is set to another value right below this call,
+	// which the original developer would likely have refactored.
+	if (m_unk0x344 != DS_NOT_A_STREAM) {
+		InvokeAction(Extra::ActionType::e_stop, m_atomId, m_unk0x344, NULL);
+		m_unk0x344 = DS_NOT_A_STREAM;
+	}
+}
+
+// FUNCTION: LEGO1 0x10025720
+// FUNCTION: BETA10 0x1006e9df
+void LegoCarBuild::FUN_10025720(undefined4 p_param)
+{
+	m_numAnimsRun++;
+	m_unk0x10a = 0;
+	MxS32 uVar6;
+
+#ifdef NDEBUG
+
+	if (GameState()->GetCurrentAct() == LegoGameState::e_act2) {
+		// This is most likely related to the helicopter rebuild in Act 2
+		switch (p_param) {
+		case 0:
+		case 1:
+		case 2:
+		case 3:
+			switch (rand() % 3) {
+			case 0:
+				m_unk0x10a = CopterScript::c_ips004d2_RunAnim;
+				StopActionIn0x344();
+				m_unk0x344 = CopterScript::c_ips004d2_RunAnim;
+				BackgroundAudioManager()->LowerVolume();
+				InvokeAction(Extra::ActionType::e_start, m_atomId, CopterScript::c_ips004d2_RunAnim, NULL);
+				break;
+			case 1:
+				m_unk0x10a = CopterScript::c_ips006d2_RunAnim;
+				StopActionIn0x344();
+				m_unk0x344 = CopterScript::c_ips006d2_RunAnim;
+				BackgroundAudioManager()->LowerVolume();
+				InvokeAction(Extra::ActionType::e_start, m_atomId, CopterScript::c_ips006d2_RunAnim, NULL);
+				break;
+			case 2:
+				m_unk0x10a = CopterScript::c_slp01xd2_RunAnim;
+				StopActionIn0x344();
+				m_unk0x344 = CopterScript::c_slp01xd2_RunAnim;
+				BackgroundAudioManager()->LowerVolume();
+				InvokeAction(Extra::ActionType::e_start, m_atomId, CopterScript::c_slp01xd2_RunAnim, NULL);
+				break;
+			}
+			break;
+		case 4:
+			FUN_10025d10(g_unk0x100d65b0[Beta0x10070520()].m_unk0x04);
+			break;
+		case 5:
+			FUN_10025d10(g_unk0x100d65b0[Beta0x10070520()].m_unk0x08);
+			break;
+		case 6:
+			m_unk0x10a = g_unk0x100d65b0[Beta0x10070520()].m_unk0x18;
+			uVar6 = m_unk0x10a;
```

Comment:
Removing `uVar6` here does decrease the match percentage, but I find its use confusing. Maybe there is an inline function that originally took an `MxS32`. Hard to tell because BETA10 does not have this part at all.